project2-beat-that/script.js

```diff
@@ -1,4 +1,299 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+var rollDice = function () {
+
+  var randomDecimal = Math.random() * 6;
+
+  var randomInteger = Math.floor(randomDecimal);
+
+  var diceNumber = randomInteger + 1;
+
+  return diceNumber;
+};
+
+//Game states 
+var INTRODUCTION = 'INTRODUCTION'
+var ROLL_DICE = 'ROLL DICE';
+var CHOOSE_DICE_ORDER = 'CHOOSE DICE ORDER';
+var CHANGE_TO_PLAYER_2 = 'CHANGE TO PLAYER 2';
+var CHOOSE_DICE_ORDER_PLAYER_2 = 'CHOOSE DICE ORDER PLAYER 2';
+var COMPARE_PLAYERS = `COMPARE PLAYERS`;
+var CONTINUE_GAME = `CONTINUE GAME`; 
+var LOWEST_COMBINE_NUMBDER = `LOWEST COMBINE NUMBDER`;
+var AUTO_COMBINE_NUMBER = `AUTO COMBINE NUMBER`
+var SCORE = `SCORE`
+
+// init gameState 
+var gameState = `INTRODUCTION`; 
+
+var playerOne = [];
+var playerTwo = [];
+var playerOneCombine = []; 
+var playerTwoCombine = []; 
+
+var playerOneRollDice = function(){
+  var playerNumber = `1`;
+  counter = 0; 
+  while (counter < 2){
+    playerOne.push(rollDice());
+    counter +=1; 
+  }
+
+  console.log(`Player One arrary is ${playerOne}`);
+  var outputMessage = `Welcome Player ${playerNumber}. <br> 
+  Your first number is: ${playerOne[0]}. <br> 
+  Your second number is ${playerOne[1]}. <br> 
+  Choose the order of your dice. <br>  <br>  Enter SCORE to check current leader. 
+  <br>  <br> Enter LOWEST to change to lowest combined number mode. `
+  return outputMessage;
+}
+
+var playerTwoRollDice = function(){
+  var playerNumber = `2`;
+  counter = 0; 
+  while (counter < 2){
+    playerTwo.push(rollDice());
+    counter +=1; 
+  }
+ 
+  console.log(`Player Two arrary is ${playerTwo}`);
+  var outputMessage = `Welcome Player ${playerNumber}. <br> 
+  Your first number is: ${playerTwo[0]}. <br> 
+  Your second number is ${playerTwo[1]}. <br> 
+  Choose the order of your dice.<br>  
+  `
+  return outputMessage;
+}
```

Comment:
repeated logic with the above function, the only change is the playerNumber which dictates whether we are pushing into playerOne or playerTwo array. consider how you can combine these two functions

Reply:
otherwise its not very scalable esp if you were to introduce >2 players